### DISCUSSION TITLE
getting actual value for floating point sort

### DISCUSSION THREAD

**Discussion by classicwuhao** [cvc5/cvc5#11469] [Q&A]
Is there an API (Java) allows us to retrieve a value for a floating point sort. I used `slv.getValue(r)` where `r` is a floating point sort defined in smt2 as follows: 
```
(assert (= r (/ 1.2 3.345)) )
```
Currently, `slv.getValue(r)` returns `(/ 80 223)`. Is it possible to get its actual value through API?

**Comment by ajreynol** [comment]
Do you mean that `r` is of type `Real`?

If so, cvc5 does not have an API for printing out a decimal approximation of a rational value. 

You may want to look at the interface for printing decimals in GMP.
